### PR TITLE
refactor gnocco into cmd + library

### PR DESCRIPTION
**Pull request by amery**
currently gnocco is a monolith relying on a global state. this PR first gets rid of the global config, and then split gnocco into cmd/gnocco for the cli and server/gnocco for the library

### CODE REVIEW

File: cmd/gnocco/server.go [hunk 1]
```diff
@@ -91,10 +101,13 @@ func (s *server) start(ds *dns.Server) {
 }
 
 func (s *server) shutDown() {
+	logger := s.cf.logger
 	logger.Info().Print("Shutdown called.")
 }
 
 func (s *server) startCacheDumping() {
+	mainconfig := s.cf
```
Comment (karasz) [inline cmd/gnocco/server.go:109]: why this line? can't you sue s.cf.Whatever instead of mainconfig.Whatever?

File: cmd/gnocco/server.go [hunk 2]
```diff
@@ -16,13 +16,17 @@ type server struct {
...
+	cf         *Gnocco
 }
 
 func (s *server) Addr() string {
 	return net.JoinHostPort(s.host, strconv.Itoa(s.port))
 }
 
 func (s *server) dumpCache() {
+	logger := s.cf.logger
+	mainconfig := s.cf
```
Comment (karasz) [inline cmd/gnocco/server.go:28]: why this line? can't you sue s.cf.Whatever instead of mainconfig.Whatever?

**Review by karasz — CHANGES_REQUESTED** [review]
Please remove the mainconfig=s.cf and use s.cf

File: cmd/gnocco/server.go (hunk 1 above)
Comment (amery) [reply cmd/gnocco/server.go:109]: very true. it was to reduce the number of changes :-/

Comment (amery) [reply cmd/gnocco/server.go:109]: otherwise  10x lines would be affected

Comment (amery) [reply cmd/gnocco/server.go:109]: I can clean that up on a follow up commit after the logic of the change is reviewed

**Review by karasz — APPROVED** [review]
LGTM

**Comment by amery** [comment]
807d99abd90fa82c577fc08f58c9d0c906526dad added to the PR. I removes the aliasing

**Review by karasz — APPROVED** [review]
Thx for implementing the cleanup.

**Pull request merged**